pyro/infer/mcmc/mcmc.py

```diff
@@ -34,16 +34,18 @@ def __init__(self, kernel, num_samples, warmup_steps=0):
     def _traces(self, *args, **kwargs):
         self.kernel.setup(*args, **kwargs)
         trace = self.kernel.initial_trace()
-        self.logger.info('Starting MCMC using kernel - {} ...'.format(self.kernel.__class__.__name__))
+        self.logger.info("Starting MCMC using kernel - {} ...".format(self.kernel.__class__.__name__))
         logging_interval = math.ceil((self.warmup_steps + self.num_samples) / 20)
         for t in range(1, self.warmup_steps + self.num_samples + 1):
             trace = self.kernel.sample(trace)
             if t % logging_interval == 0:
-                self.logger.info('Iteration: {}.'.format(t))
+                self.logger.info("Iteration: {}.".format(t))
                 diagnostic_info = self.kernel.diagnostics()
                 if diagnostic_info is not None:
                     self.logger.info(diagnostic_info)
             if t <= self.warmup_steps:
+                if t == self.warmup_steps:
+                    self.kernel._warmup = False
```

Comment:
If we are accessing the `._warmup` kernel attribute, we might as well rename it to `.is_warmup`.

Reply:
@neerajprad I think that it is better to have a method to do it. What do you think about: `end_warmup`?

Reply:
Yeah, making `end_warmup` method is also useful to set step_size at the end of DA scheme (it takes the average of previous step_sizes). Previously, I have to make `warmup` and `adapted` flag to deal with that logic.